docs/table_design/expression_partitioning.md

```diff
@@ -0,0 +1,256 @@
+| 17113       | p20230227     | 2              | 2023-07-19 17:53:59 | 0                  | NORMAL | event_day    | [types: [DATETIME]; keys: [2023-02-27 00:00:00]; ..types: [DATETIME]; keys: [2023-02-28 00:00:00]; ) | event_day, site_id | 6       | 3              | HDD           | 9999-12-31 23:59:59 | NULL                     | 0B       | false      | 0        |
++-------------+---------------+----------------+---------------------+--------------------+--------+--------------+------------------------------------------------------------------------------------------------------+--------------------+---------+----------------+---------------+---------------------+--------------------------+----------+------------+----------+
+2 rows in set (0.00 sec)
+```
+
+Example 2: If you want to implement partition lifecycle management, which is to retain only a certain number of recent partitions and delete historical partitions, you can use the `partition_live_number` property to specify the number of partitions to retain.
+
+```SQL
+CREATE TABLE site_access2 (
+    event_day DATETIME NOT NULL,
+    site_id INT DEFAULT '10',
+    city_code VARCHAR(100),
+    user_name VARCHAR(32) DEFAULT '',
+    pv BIGINT DEFAULT '0'
+) 
+DUPLICATE KEY(event_day, site_id, city_code, user_name)
+PARTITION BY date_trunc('month', event_day)
+DISTRIBUTED BY HASH(event_day, site_id)
+PROPERTIES(
+    "partition_live_number" = "3" -- only retains the most recent three partitions
+);
+```
+
+Example 3: Suppose you frequently query data by week.  You can use the partition expression `time_slice()` and set the partition column as `event_day` and the partition granularity to seven days at table creation. Data of one week is stored in one partition and partition pruning can be used to significantly improve query efficiency.
+
+```SQL
+CREATE TABLE site_access(
+    event_day DATETIME NOT NULL,
+    site_id INT DEFAULT '10',
+    city_code VARCHAR(100),
+    user_name VARCHAR(32) DEFAULT '',
+    pv BIGINT DEFAULT '0'
+)
+DUPLICATE KEY(event_day, site_id, city_code, user_name)
+PARTITION BY time_slice(event_day, INTERVAL 7 day)
+DISTRIBUTED BY HASH(event_day, site_id)
+```
+
+## Partitioning based on the column expression (since v3.1)
+
+If you frequently query and manage clearly categorized data based on specific types, you only need tospecify the column representing the type as the partition column. StarRocks will automatically create partitions based on the partition column values of the loaded data.
+
+However, in some special scenarios, such as  when the table contains a column `city`, and you frequently query and manage data based on countries and cities. You must use [list partitioning](./list_partitioning.md) to store data of multiple cities within the same country in one partition.
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/0f76a6a6-7b95-4d97-84a0-424ac913be20)
double spaces. Check the entire docs to see if there are any other double spaces that require attention.